projects/packages/forms/src/blocks/contact-form/editor.scss

```diff
@@ -42,69 +42,111 @@
-		display: flex;
-		flex-wrap: wrap;
-		justify-content: flex-start;
-		flex-direction: row;
-		gap: var(--wp--style--block-gap, 1.5rem);
-
-		.wp-block {
-			flex: 0 0 100%;
-			margin-top: 0;
-			margin-bottom: 0;
+		// In a constrained layout, the blocks take up a 'content size' that's defined by the theme or user.
+		// Blocks are centered using `margin: auto;`.
+		// The percentage widths need to be multiplied by the content size and additional right margin is used to keep
+		// the blocks aligned to the left of the container.
+		// Field block currently don't support alignment, so only the default 'left' alignment is supported.
+		// TODO: consider RTL.
+		&.is-layout-constrained > * {
+			&.jetpack-field__width-25 {
+				max-width: calc(var(--wp--style--global--content-size) * 0.25);
+				margin-right: calc(50% + (var(--wp--style--global--content-size) * 0.25)) !important;
+			}
 
-			&.wp-block-jetpack-button {
-				flex-basis: auto;
-				align-self: flex-end; // Ensure button is aligned with the bottom of the previous field when on a same row
+			&.jetpack-field__width-50 {
+				max-width: calc(var(--wp--style--global--content-size) * 0.5);
+				margin-right: 50% !important;
 			}
 
-			&.jetpack-field__width-25,
-			&.jetpack-field__width-50,
 			&.jetpack-field__width-75 {
-				box-sizing: border-box;
+				max-width: calc(var(--wp--style--global--content-size) * 0.75);
+				margin-right: calc(50% - (var(--wp--style--global--content-size) * 0.25)) !important;
 			}
+		}
 
+		// In a flow layout, the blocks take up the full size of the container, so simple percentage widths work.
+		&.is-layout-flow > * {
 			&.jetpack-field__width-25 {
-				flex: 1 1 calc( 25% - calc(var(--wp--style--block-gap, 1.5rem) * 1) );
 				max-width: 25%;
-
-				.jetpack-option__input.jetpack-option__input.jetpack-option__input {
-					width: 70px;
-				}
 			}
 
 			&.jetpack-field__width-50 {
-				flex: 1 1 calc( 50% - calc(var(--wp--style--block-gap, 1.5rem) * 1) );
 				max-width: 50%;
 			}
 
 			&.jetpack-field__width-75 {
-				flex: 1 1 calc( 75% - calc(var(--wp--style--block-gap, 1.5rem) * 1) );
 				max-width: 75%;
 			}
+		}
 
-			@media (max-width: #{ ($break-mobile) }) {
-				&[class*=jetpack-field__width-] {
-					flex-basis: 100%;
-					max-width: none;
+		// When there's no layout set on the parent block, use the form's own layout rules.
+		// This is a flexbox 'row' layout where the fields are usually 100% width and wrap
+		// to the next line.
+		&:not(.is-layout-constrained, .is-layout-flex, .is-layout-grid, .is-layout-flow)  {
+			display: flex;
+			flex-wrap: wrap;
+			justify-content: flex-start;
+			flex-direction: row;
+			gap: var(--wp--style--block-gap, 1.5rem);
+
+			> * {
+				flex: 0 0 100%;
+				margin-top: 0;
+				margin-bottom: 0;
+
+				&.wp-block-jetpack-button {
+					flex-basis: auto;
+					align-self: flex-end; // Ensure button is aligned with the bottom of the previous field when on a same row
```

Comment:
Oh this works now! Previously the styles weren't applying I think due to being nested under the wrong classname.

Reply:
Oh right, this is the git conflict that I solved. It was introduced in this PR - https://github.com/Automattic/jetpack/pull/41576, so I didn't do much!

Reply:
Ah, I see what you mean now, bringing that change in has fixed the issues I have with the button width. Nice 🎉